sklearn/utils/estimator_checks.py

```diff
@@ -4083,7 +4083,13 @@ def check_param_validation(name, estimator_orig):
 
         for method in methods:
             with raises(ValueError, match=match, err_msg=err_msg):
-                getattr(estimator, method)(X, y)
+                method_to_call = getattr(estimator, method)
+                method_signature = signature(method_to_call)
+                if "X" not in method_signature.parameters:
```

Comment:
I'm on the fence about using this heuristic. What do you think about something more structured like the tags?

```python
any(X_type.endswith("labels") for X_type in safe_tags(estimator, key="X_types")
```

Reply:
Fine with me.